### PR TITLE
Initialize part of the process of creating a virtual control plane

### LINKED ISSUE #421: feature: k8s in k8s
**What would you like to be added**:
Deploy kosmos through virtual control plane to implement tenant management

**proposal**: https://kosmos-io.github.io/website/v0.2.0/proposals/k8s-in-k8s

**Why is this needed**:

### PR DESCRIPTION
**Pull request by yuleichun-striving**
#### What type of PR is this?
/kind feature

#### What does this PR do?
Initialize part of the process of creating a virtual control plane

#### Which issue(s) does this PR fix?

Fixes #421
#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by yuleichun-striving** [comment]
@cc @duanmengkk @GreatLazyMan

File: examples/vitual-cluster-demo.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  namespace: test
+  name: test
+spec:
+  promoteResources:
+    nodes:
+      - node1
+      - node2
+    resources:
+      cpu: 1
+      memory: "1Gi"
```
Comment (duanmengkk) [inline examples/vitual-cluster-demo.yaml:13]: add a new line

**Pull request merged**